waiter/src/waiter/core.clj

```diff
@@ -927,6 +930,18 @@
                                       scheduler instance-rpc-chan quanta-constraints scaling-timeout-config
                                       scheduler-interactions-thread-pool service-id))
                                   {})))
+   :codahale-reporters (pc/fnk [[:settings [:metrics-config reporters]]]
+                         (pc/map-vals
+                           (fn make-codahale-reporter [{:keys [factory-fn] :as reporter-config}]
```

Comment:
Nitpick: Can we move `make-codahale-reporter` to `waiter.reporter`? That way we do not need the `(waiter.reporter CodahaleReporter)` import.